pkg/auth/login_credentials_manager.go

```diff
@@ -260,6 +295,19 @@ func (h *LoginCredentialsManagerImpl) getNetrcMachine(filterParams netrc.NetrcMa
 	return netrcMachine, err
 }
 
+func (h *LoginCredentialsManagerImpl) GetOnPremSsoCredentials(url, caCertPath string, unsafeTrace bool) func() (*Credentials, error) {
+	return func() (*Credentials, error) {
+		isSSO := h.isOnPremSSOUser(url, caCertPath, unsafeTrace)
+		if isSSO {
```

Comment:
Do we necessarily need this? I think if you return the credential struct and SSO is false, it doesn't count that as a "complete" credential set

Reply:
Good point